Backend/src/test/java/ca/mcgill/cooperator/controller/CourseOfferingControllerIT.java

```diff
@@ -0,0 +1,171 @@
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+public class CourseOfferingControllerIT extends ControllerIT {
+
+    @Autowired private MockMvc mvc;
+
+    @Autowired private ObjectMapper objectMapper;
+
+    @Autowired CourseRepository courseRepository;
+    @Autowired CourseOfferingRepository courseOfferingRepository;
+    @Autowired CourseOfferingService courseOfferingService;
+
+    @BeforeEach
+    @AfterEach
+    public void clearDatabase() {
+        courseOfferingRepository.deleteAll();
+        courseRepository.deleteAll();
+    }
+
+    /**
+     * Tests creating, reading, updating and deleting a course
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testCourseOfferingFlow() throws Exception {
+        // 1. create the Course with the helper method
+        CourseDto courseDto = createTestCourse();
+
+        // 2. create the CourseOffering with a post request
+        CourseOfferingDto courseOfferingDto = new CourseOfferingDto();
+        courseOfferingDto.setYear(2020);
+        courseOfferingDto.setSeason(Season.WINTER);
+        courseOfferingDto.setCourse(courseDto);
+
+        MvcResult mvcResult =
+                mvc.perform(
+                                post("/course-offerings")
+                                        .contentType(MediaType.APPLICATION_JSON)
+                                        .content(objectMapper.writeValueAsString(courseOfferingDto))
+                                        .characterEncoding("utf-8"))
+                        .andExpect(status().isOk())
+                        .andReturn();
+
+        // get object from response
+        courseOfferingDto =
+                objectMapper.readValue(
+                        mvcResult.getResponse().getContentAsString(), CourseOfferingDto.class);
+
+        // 2. get the courseOffering by ID, valid
+        mvc.perform(
+                        get("/course-offerings/" + courseOfferingDto.getId())
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        // 3. test getting all CourseOfferings
+        mvcResult =
+                mvc.perform(get("/course-offerings").contentType(MediaType.APPLICATION_JSON))
+                        .andExpect(status().isOk())
+                        .andReturn();
+
+        // get object from response
+        List<CourseOfferingDto> returnedCourseOfferings =
+                Arrays.asList(
+                        objectMapper.readValue(
+                                mvcResult.getResponse().getContentAsString(),
+                                CourseOfferingDto[].class));
+
+        assertEquals(returnedCourseOfferings.size(), 1);
```

Comment:
make this equals more through, maybe compare the name of the courseoffering or the Id to ensure its robust